bundles/binding/org.openhab.binding.tacmi/src/main/java/org/openhab/binding/tacmi/internal/TACmiBinding.java

```diff
@@ -0,0 +1,298 @@
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.Map;
+
+import org.openhab.binding.tacmi.TACmiBindingProvider;
+import org.openhab.binding.tacmi.internal.message.AnalogMessage;
+import org.openhab.binding.tacmi.internal.message.AnalogValue;
+import org.openhab.binding.tacmi.internal.message.DigitalMessage;
+
+import org.apache.commons.lang.StringUtils;
+import org.openhab.core.binding.AbstractActiveBinding;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implement this class if you are going create an actively polling service like
+ * querying a Website/Device.
+ * 
+ * @author Timo Wendt
+ * @since 1.7.0
+ */
+public class TACmiBinding extends AbstractActiveBinding<TACmiBindingProvider> {
+
+	private static final Logger logger = LoggerFactory
+			.getLogger(TACmiBinding.class);
+
+	/**
+	 * The BundleContext. This is only valid when the bundle is ACTIVE. It is
+	 * set in the activate() method and must not be accessed anymore once the
+	 * deactivate() method was called or before activate() was called.
+	 */
+	private BundleContext bundleContext;
+
+	/**
+	 * the refresh interval which is used to poll values from the TACmi server
+	 * (optional, defaults to 60000ms)
+	 */
+	private long refreshInterval = 1000;
+
+	/**
+	 * IP or hostname of the CMI This is set in the activate method
+	 */
+	private String cmiAddress;
+
+	/**
+	 * Port the cmi is listening on and also sending to
+	 */
+	private static int cmiPort = 5441;
+
+	/**
+	 * Connection socket
+	 */
+	private DatagramSocket clientSocket = null;
+
+	public TACmiBinding() {
+	}
+
+	/**
+	 * Called by the SCR to activate the component with its configuration read
+	 * from CAS
+	 * 
+	 * @param bundleContext
+	 *            BundleContext of the Bundle that defines this component
+	 * @param configuration
+	 *            Configuration properties for this component obtained from the
+	 *            ConfigAdmin service
+	 */
+	public void activate(final BundleContext bundleContext,
+			final Map<String, Object> configuration) {
+		this.bundleContext = bundleContext;
+
+		// the configuration is guaranteed not to be null, because the component
+		// definition has the
+		// configuration-policy set to require. If set to 'optional' then the
+		// configuration may be null
+
+		// to override the default refresh interval one has to add a
+		// parameter to openhab.cfg like <bindingName>:refresh=<intervalInMs>
+		String refreshIntervalString = (String) configuration.get("refresh");
+		if (StringUtils.isNotBlank(refreshIntervalString)) {
+			refreshInterval = Long.parseLong(refreshIntervalString);
+		}
+
+		// set cmiAddress from configuration
+		cmiAddress = (String) configuration.get("cmiAddress");
```

Comment:
i would assume the binding won't work without the cmiAddress, right? If yes, please check the validity of the configuration and write some WARN logs if the configuration is invalid.